build-tests/api-documenter-test/etc/api-documenter-test.api.json

```diff
@@ -1017,6 +1047,7 @@
             },
             {
               "kind": "IndexSignature",
+              "canonicalReference": "api-documenter-test!IDocInterface3#:signature(0)",
```

Comment:
@rbuckton The `CallSignature`, `ConstructSignature`, and `IndexSignature` all get assigned the same UID here.

How do we avoid this UID conflict?

Reply:
Hmm. I was under the impression these weren't yet fully supported in api-extractor. In TypeScript they each have their own symbol name of `"__call"`, `"__new"`, and `"__index"`, but they all have the meaning of `SymbolFlags.Signature`.

Reply:
But what if someone declares a regular member called `__call`?  ([Example](http://www.typescriptlang.org/play/#code/KYDwDg9gTgLgBAYwDYEMDOa4A04G8BQciEAdmjFAK4IzQAUAlHoXAL4sD6HJwA7gFxwSlALYAjYFDgBeOAEYA3Jw4BLEgBNQg8lDUBzGXADkII0qJcEKJEkFiIEJMBQlDAM2tpgS9kA))

Reply:
In TypeScript we internally reserve double-underscore prefixed strings and escape them by prefixing an extra underscore:

```ts
interface Foo {
 (): void; // has the symbol name "__call"
 __call(): void; // has the symbol name "___call"
}
```

Reply:
As far as I know though, this escaping is an internal implementation detail of the compiler, and somewhat careful accounting is needed to keep track of whether a string is escaped or not.  

In the language, any possible sequence of printable characters is allowed as an identifier, so I prefer to think of signatures as not having any name.

Reply:
I think this might be a better approach: https://github.com/microsoft/web-build-tools/pull/1406#discussion_r305651761

Reply:
Support for `:call`, ':new', and ':index' added in https://github.com/microsoft/tsdoc/pull/173